spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/LogLevels.java

```diff
@@ -57,4 +58,11 @@
 	 */
 	String level() default "DEBUG";
 
+	/**
+	 * When used at the class level, set this to PER_CLASS to set/restore
+	 * the levels once only. Useful with {@code @RepeatedTest}.
+	 * @return the Lifecycle.
+	 */
+	TestInstance.Lifecycle lifecycle() default TestInstance.Lifecycle.PER_METHOD;
```

Comment:
Can't this be per class by default? In most cases we really use it on the class level without considering it to be changed in between. Well, essentially it does just an idempotent work between methods anyway, so, I even would consider do not have this option at all. On class level it is for the whole class; on method level - just for method and has a precedence over the class one.

What am I missing with this option, please?

Thanks

Reply:
I kept it this way so we get the nice delimiter in the log between tests; but I have another idea. Stay tuned...